src/libraries/Microsoft.Extensions.Configuration.CommandLine/src/CommandLineConfigurationProvider.cs

```diff
@@ -39,96 +39,159 @@ public CommandLineConfigurationProvider(IEnumerable<string> args, IDictionary<st
         public override void Load()
         {
             var data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-            string key, value;
 
             using (IEnumerator<string> enumerator = Args.GetEnumerator())
             {
+                bool isNext = enumerator.MoveNext();
+
+                // Store 1st argument here and start while loop with the 2nd,
+                // or 'Current', argument. This is so we can look at the
+                // 'Current' argument in relation to the 1st while evaluating
+                // the 1st.
+                string previousArg = enumerator.Current;
+
+                // If no first arg, return empty dictionary
+                if (!isNext || previousArg == null)
+                {
+                    Data = data;
+                    return;
+                }
+
                 while (enumerator.MoveNext())
                 {
+                    // 'Current' is now the 2nd argument in Args
                     string currentArg = enumerator.Current;
-                    int keyStartIndex = 0;
-
-                    if (currentArg.StartsWith("--"))
+                    // TryProcessArgs will return false if previousArg is invalid
+                    if (TryProcessArgs(previousArg, currentArg, out (string key, string value) loopPair))
                     {
-                        keyStartIndex = 2;
-                    }
-                    else if (currentArg.StartsWith("-"))
-                    {
-                        keyStartIndex = 1;
-                    }
-                    else if (currentArg.StartsWith("/"))
-                    {
-                        // "/SomeSwitch" is equivalent to "--SomeSwitch" when interpreting switch mappings
-                        // So we do a conversion to simplify later processing
-                        currentArg = string.Format("--{0}", currentArg.Substring(1));
-                        keyStartIndex = 2;
+                        // Override value when key is duplicated,
+                        // so we always have the last argument win.
+                        data[loopPair.key] = loopPair.value;
                     }
+                    previousArg = enumerator.Current;
+                }
 
-                    int separator = currentArg.IndexOf('=');
+                // Process the last previousArg after exiting loop
+                if (TryProcessArgs(previousArg, null, out (string key, string value) lastPair))
```

Comment:
Wouldn't this be calling `TryProcessArgs` with the last arg twice? `previousArg` will be pointing to `enumerator.Current` which is the same as `currentArg` if it entered the while loop.

Reply:
The variable `previousArg` gets assigned the value of `currentArg` at the end of the last loop in the `while` block. When it reaches the point where it processes the last argument (outside the while loop), `previousArg` and `currentArg` are indeed the same value, but that value has not yet been passed as the `previousArg` to the `TryProcessArgs` method. It only goes through it once.